Apply reconfig guard records even if setting is disabled

RameshIyyar opened the pull request.

ojayanth commented:
Please update commit messages with additional info , basically you are chnaging existing flow , need to say why this is doing now. 
Also rebase te commit.

RameshIyyar commented:
> Please update commit messages with additional info , basically you are chnaging existing flow , need to say why this is doing now. Also rebase te commit.

Added more details and rebased it with latest patches.
